src/snowflake/connector/result_batch.py

```diff
@@ -72,21 +72,25 @@ def create_batches_from_response(
     cursor: "SnowflakeCursor",
     _format: str,
     data: Dict[str, Any],
+    schema: List["ResultMetadata"],
 ) -> List["ResultBatch"]:
     column_converters: List[Tuple[str, "SnowflakeConverterType"]] = []
     arrow_context: Optional["ArrowConverterContext"] = None
     rowtypes = data["rowtype"]
-    column_names: List[str] = [c["name"] for c in rowtypes]
     total_len: int = data.get("total", 0)
     first_chunk_len = total_len
     rest_of_chunks: List["ResultBatch"] = []
     if _format == "json":
-        column_converters: List[Tuple[str, "SnowflakeConverterType"]] = [
-            (
-                c["type"],
-                cursor._connection.converter.to_python_method(c["type"].upper(), c),
+
+        def col_to_converter(col):
```

Comment:
If you add a function, don't forget to type-hint it :D
```suggestion
        def col_to_converter(col -> Dict[str, Any]) -> Tuple[str, "SnowflakeConverterType"]:
```

Again, I'm not 100% on the key type-hint, but I think `str` is the only thing it can be